src/XrdHttp/XrdHttpError.hh

```diff
@@ -0,0 +1,79 @@
+    private:
+        // HTTP response parameters to be sent back to the user
+        int httpStatusCode;
+        std::string httpErrorCode;
+        std::string httpStatusText;
+        const char *desc;
+        const char *header_to_add;
+
+        static std::string formatStatusText(std::string errCode, std::string etext) {
+        return ("ERROR: " + errCode + ": " + etext);
+        }
+
+    
+    public:
+        XrdHttpError(int code, const std::string &errCode, const std::string &statusText, const char *desc, const char *header_to_add)
+        : httpStatusCode(code), httpErrorCode(errCode), httpStatusText(statusText), desc(desc), header_to_add(header_to_add) {}
+
+
+        static XrdHttpError Create( XResponseType xrdresp, XErrorCode xrderrcode, XrdHttpReq::ReqType httpVerb, XRequestTypes xrdOperation, std::string etext, const char *desc, const char *header_to_add){
+            int code{0};
+            std::string errCode;
+            std::string statusText;
+
+            switch (httpVerb){
+                case XrdHttpReq::rtPUT: 
+                    if (xrdOperation == kXR_open){
+                        if (xrderrcode == kXR_isDirectory) {
+                            code = 409;
+                            errCode = "8.1";
+                        }
+                        else if (xrderrcode == kXR_NoSpace) {
+                            code = 507;
+                            errCode = "8.3.1";
+                        }
+                        else if (xrderrcode == kXR_overQuota){
+                            code = 507;
+                            errCode = "8.3.2";
+                        }
+                        else if (xrderrcode == kXR_NotAuthorized){
+                            code = 403;
+                            errCode = "9.3";
+                        }
+                    }
+                    else if (xrdOperation == kXR_write){
+                        if (xrderrcode == kXR_NoSpace) {
+                            code = 507;
+                            errCode = "8.4.1";
+                        }
+                        else if (xrderrcode == kXR_overQuota){
+                            code = 507;
+                            errCode = "8.4.2";
+                        }
+                    }
+                    else {
+                        code = 500;
+                        errCode = "Unknown";
+                    }
+                    break;
+                default:
+                    code = 500;
+                    errCode = "Internal Server Error";
+                    break;
+            }
+            statusText = formatStatusText(errCode, etext);
+            return XrdHttpError(code, errCode, statusText, desc, header_to_add);
+        }
+
+        int Send(XrdHttpProtocol *prot, bool keepalive){
+            return prot->SendSimpleResp(httpStatusCode, desc, header_to_add,
+                httpStatusText.c_str(), httpStatusText.length(), keepalive);
```

Comment:
The usage of this class seems to be always to create the error, call this, then destroy the object. I think the idea of just using a function for the combined operation taking all parameters and returning in one go might be cleaner.